packages/core/src/core/adapter_format.ts

```diff
@@ -332,6 +332,34 @@ export const calculatePatch = async ({
   }
 }
 
+const fixAdditionPaths = async (workspace: Workspace, changes: ReadonlyArray<Change>): Promise<Change<Element>[]> => {
+  const additionElements = changes.filter(isAdditionChange).map(getChangeData)
+  const unmergedAdditionElements = await unmergeElements(workspace, workspace.currentEnv(), additionElements)
+  const unmergedElementsById = _.groupBy(unmergedAdditionElements, elem => elem.elemID.getFullName())
+
+  const fixAddition = (change: Change): Change<Element>[] =>
+    unmergedElementsById[getChangeData(change).elemID.getFullName()].map(elem => toChange({ after: elem }))
+
+  return changes.flatMap(change => (isAdditionChange(change) ? fixAddition(change) : [change]))
+}
+
+const updateToWorkspace = async (args: {
+  workspace: Workspace
+  toWorkspace: Workspace | undefined
+  changes: ReadonlyArray<Change>
+}): Promise<ReadonlyArray<Change>> => {
+  const { workspace, toWorkspace, changes } = args
```

Comment:
Nit (preference, take it or leave it):
```suggestion
const updateToWorkspace = async ({
  workspace,
  toWorkspace,
  changes
}: {
  workspace: Workspace
  toWorkspace: Workspace | undefined
  changes: ReadonlyArray<Change>
}): Promise<ReadonlyArray<Change>> => {
```